### PR TITLE
Make Style nullable

### LINKED ISSUE #25227: Entry.Style is Not Nullable
### Description

`(new Entry()).Style = null;` compiles without error in .Net 8 and earlier but in .NET 9 it produces a warning `warning CS8625: Cannot convert null literal to non-nullable reference type.`

### Steps to Reproduce

1. Create a new MAUI App
2. Add `(new Entry()).Style = null;` somewhere (I did it in `OnCounterClicked`, but anywhere syntactically correct will do.
3. On .NET 9 it will emit the warning on 8 it will compile.

I did this on .NET 9 RC2 but reported RC1 below because there's no pulldown value for RC2 yet.

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.0-rc.1.24453.9

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.91 SR9.1

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

all

### Did you find any workaround?

Do not treat warnings as errors and just ignore the warning.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by pictos**
### Description of Change

Add the `?` symbol to the `Style` property of controls.

### Issues Fixed

Fixes #25227

### CODE REVIEW

File: src/Controls/src/Core/PublicAPI/net-android/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@
 *REMOVED*~Microsoft.Maui.Controls.NavigableElement.Style.set -> void
 *REMOVED*~Microsoft.Maui.Controls.NavigableElement.StyleClass.get -> System.Collections.Generic.IList<string>
 *REMOVED*~Microsoft.Maui.Controls.NavigableElement.StyleClass.set -> void
+Microsoft.Maui.Controls.StyleableElement.Style.get -> Microsoft.Maui.Controls.Style?
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/PublicAPI/net-android/PublicAPI.Unshipped.txt:12]: This change un PublicAPIs is pending in Tizen:

```
D:\a\_work\1\s\src\Controls\src\Core\StyleableElement\StyleableElement.cs(25,4): error RS0016: Symbol 'Style.get' is not part of the declared public API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-tizen7.0]
D:\a\_work\1\s\src\Controls\src\Core\PublicAPI\net-tizen\PublicAPI.Unshipped.txt(86,1): error RS0017: Symbol 'Microsoft.Maui.Controls.StyleableElement.Style.get -> Microsoft.Maui.Controls.Style!' is part of the declared API, but is either not public or could not be found (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-tizen7.0]
    22 Warning(s)
    2 Error(s)
```

**Review by jsuarezruiz — DISMISSED** [review]

**Comment by StephaneDelcroix** [comment]
it feels weird that among all the properties we have, not attributed as nullable, that Style is the only one to fail.

**Comment by mattleibow** [comment]
I figured it out. We forgot to add the `#nullable disable` when we moved the style code into the new `StyleableElement`.

**Review by mattleibow — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Review by mattleibow — DISMISSED** [review]

**Review by PureWeen — DISMISSED** [review]
/Users/builder/azdo/_work/3/s/src/Controls/src/Core/PublicAPI/net-tizen/PublicAPI.Unshipped.txt(88,1): error RS0017: Symbol 'Microsoft.Maui.Controls.StyleableElement.Style.get -> Microsoft.Maui.Controls.Style!' is part of the declared API, but is either not public or could not be found (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [/Users/builder/azdo/_work/3/s/src/Controls/src/Core/Controls.Core.csproj::TargetFramework=net9.0-tizen7.0]

**Comment by pictos** [comment]
@PureWeen I removed that entry, not sure if I understood how that API file is written, but I believe when there's a `!` at the end, it indicates that value is not nullable, so I remove that line, since there's the same symbol declared there, but it's nullable

**Review by jfversluis — DISMISSED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**